numba/tests/test_repr.py

```diff
@@ -0,0 +1,53 @@
+import unittest
+import numpy as np
+
+from numba.tests.support import TestCase
+from numba import typeof
+from numba.core import types
+from numba.typed import List, Dict
+
+NB_TYPES = [
+    types.Array, types.NestedArray, types.int64, types.float64,
+    types.unicode_type, types.Record, types.UnicodeCharSeq, types.UniTuple,
+    types.List, types.Tuple, types.DictType, types.ListType, types.Set,
+    types.bool_
+]
+
+
+class TestRepr(TestCase):
+    def check_repr(self, val):
+        ty = typeof(val)
+        tys_ns = {ty.__name__: ty for ty in NB_TYPES if hasattr(ty, "__name__")}
+        tys_ns.update({ty.name: ty for ty in NB_TYPES if hasattr(ty, "name")})
```

Comment:
This is invariant to `check_repr` and can be moved to `setUp`:

```python
class TestRepr(TestCase):
    def setUp(self) -> None:
        tys_ns = {ty.__name__: ty for ty in NB_TYPES if hasattr(ty, "__name__")}
        tys_ns.update({ty.name: ty for ty in NB_TYPES if hasattr(ty, "name")})
        self.tys_ns = tys_ns
        return super().setUp()

    def check_repr(self, val):
        ty = typeof(val)
        ty2 = eval(repr(ty), self.tys_ns)
        self.assertEqual(ty, ty2)
```

Reply:
good idea, I've added it now.